core/application/impl/chain_spec_impl.cpp

```diff
@@ -162,16 +163,38 @@ namespace kagome::application {
     if (code_substitutes_opt.has_value()) {
       for (const auto &[hash, code] : code_substitutes_opt.value()) {
         OUTCOME_TRY(hash_processed, common::Hash256::fromHexWithPrefix(hash));
-        OUTCOME_TRY(code_processed, common::unhexWith0x(code.data()));
-        // TODO(sanblch): move this from memory to db
-        // https://github.com/soramitsu/kagome/issues/935
-        code_substitutes_->emplace(hash_processed, code_processed);
+        // OUTCOME_TRY(code_processed, common::unhexWith0x(code.data()));
+        code_substitutes_->emplace(hash_processed);
       }
     }
 
     return outcome::success();
   }
 
+  outcome::result<common::Buffer> ChainSpecImpl::fetchCodeSubstituteByHash(
+      const common::Hash256 hash) const {
+    pt::ptree tree;
+    try {
+      pt::read_json(config_path_, tree);
```

Comment:
Hm... 
From one hand side, if config_path_ is initialized, it means json already parsed at least once (in loadFromJson), and extra work happens.
From other hand side, if config_path_ is defined, this will throw an error.
Think about refactor that, to avoid non initialized path or doing same work twice.

Reply:
Well, yes - it is double work - since it's a lazy loading for a single particular value. Storing a tree is not a good decision here.
Please, elaborate what exactly you find problematic here?

Reply:
Okay, added a check for a premature stop if hash is not expected to be in a file.